### ISSUE TITLE
How to store gene symbols and ids together in a seurat object?

### ISSUE THREAD

**Issue by partrita** [satijalab/seurat#8673]
I found that when I used the `Read10x()` function to load the pbmc3k dataset that only gene symbol data was available in the Seurat object.

While it's possible to generate data from existing gene symbols using `biomart`, I would prefer to preserve the ENSEMBL ID from the beginning. 

Because I want to recreate the data in the original 10x format(mtx, genes.tsv, barcode.tsv)from seurat object.

```R
# Load the PBMC dataset
pbmc_data <- Read10X(data.dir = "../input/filtered_gene_bc_matrices/hg19/")
genes_id <- read.table(
    "../input/filtered_gene_bc_matrices/hg19/genes.tsv",
    header=FALSE,
    sep="\t",
    )
# Initialize the Seurat object with the raw (non-normalized data).
pbmc <- CreateSeuratObject(
    counts = pbmc_data,
    project = "pbmc3k"
    )
pbmc
```

```R
# Retrieve data in an expression matrix RNA counts matrix
matrix <- pbmc[["RNA"]]$counts

# View metadata data frame, stored in object@meta.data
meta_data <- pbmc[[]]
genes<- rownames(pbmc)
barcode <- colnames(pbmc)
```
In other words, is there a way to split the rownames, which represent the list of genes, into two layers for usage? Or is using biomart the best approach?

Thanks

**Comment by samuel-marsh** [comment]
Hi,

Not member of dev team but hopefully can be helpful.  One solution is function I have added to my package scCustomize (currently use the develop branch on GitHub not CRAN release for this function).

This will create a feature ID table in the @misc slot of the object.  It will contain only symbols found in the supplied assay.

On biomart note that will have issue because of re-naming the Seurat enforces.  There are handful of genes that have multiple ensembl ids.  Seurat by default makes these unique using base R `make.unique` which appends extension (.1, .2 , etc) on the end of non-unique entries.  So if you were to feed list into biomart some of the genes would not be correctly found.

There may be better solution but this is just what I thought of.

Best,
Sam

**Comment by partrita** [comment]
> Hi,
> 
> Not member of dev team but hopefully can be helpful. One solution is function I have added to my package scCustomize (currently use the develop branch on GitHub not CRAN release for this function) called `Add_Alt_Feature_ID`.
> 
> This will create a feature ID table in the @misc slot of the object. It will contain only symbols found in the supplied assay.
> 
> On biomart note that will have issue because of re-naming the Seurat enforces. There are handful of genes that have multiple ensembl ids. Seurat by default makes these unique using base R `make.unique` which appends extension (.1, .2 , etc) on the end of non-unique entries. So if you were to feed list into biomart some of the genes would not be correctly found.
> 
> There may be better solution but this is just what I thought of.
> 
> Best, Sam

Oh, Thanks for the tips! I will try `scCustomize`.

To be clear, you created a new tool yourself because you couldn't find the that kind of function, right?. I admire your initiative in your work.

Best,
